service/src/test/java/bio/terra/externalcreds/controllers/OidcApiControllerTest.java

```diff
@@ -0,0 +1,67 @@
+package bio.terra.externalcreds.controllers;
+
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import bio.terra.externalcreds.BaseTest;
+import bio.terra.externalcreds.services.ProviderService;
+import java.util.List;
+import java.util.Set;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+
+public class OidcApiControllerTest extends BaseTest {
+
+  @Autowired private MockMvc mvc;
+
+  @MockBean private ProviderService providerService;
+
+  @Test
+  void testListProviders() throws Exception {
+    when(providerService.getProviderList()).thenReturn(Set.of("fake-provider2", "fake-provider1"));
+
+    mvc.perform(get("/api/oidc/v1/providers"))
+        .andExpect(content().json("[\"fake-provider1\",\"fake-provider2\"]"));
+  }
+
+  @Test
+  void testGetAuthUrl() throws Exception {
+    String result = "https://test/authorization/uri";
+    String provider = "fake";
+    String redirectUri = "fakeuri";
+    Set<String> scopes = Set.of("openid", "email");
+    String state = null;
```

Comment:
Could we make these `public final` class-level things so we don't have to redefine them for every test? If the queryParams are going to be the same for each test, we could also make a shared `MultiValueMap` and initialize it in `@beforeEach`

Reply:
I understand the desire to do this however this leads to coupled tests that really have nothing to do with each other. Then down the road we have a chunk of test data that feeds a bunch of different tests making it very difficult to change individual tests or the data.

Reply:
makes sense!